### PR TITLE
[v0.7.0] add email sender and webmail

### PR DESCRIPTION
**Pull request by freignat91**
Related to #748

Do not merge before #756 and #754

Email sender provide function to send plain test or html email using templates and high level functions to send specific emails as:

SendAccountVerificationEmail(to string, accountName string, token string) error

SendAccountResetPasswordEmail(to, string, accountName string, token string) error

SendAccountPasswordConfirmationEmail(to string, accountName string)

these email answers are managed by webmail module which receive the answer, execute the right appcelerator account functions and send back htrml page considering the account functions answers.

hack/amptools make -f Makefile.refactor.make cleanall install-deps
make install
make test

email design has to be validated, especially the pictures.

for theses emails, i used recommendations from:

https://postmarkapp.com/blog/password-reset-templates-best-practices
http://99rabbits.com/email-sign-up-confirmation/
https://reallygoodemails.com/category/inaugural/account/
https://reallygoodemails.com/category/password-reset/

### CODE REVIEW

File: pkg/ampmail/accountPasswordConfirm.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
...BS34T/sVfDjwb4o8ZaxeSaf8AEC7jtNAj0uza8uNRLx+YHSNcHYFKZbsZEHevdNKv/wC1NMt7ryZ7f7REsvlTLtkj3AHaw7MM4I7Gps9xliiiigAr8sf+DhX9pm6+EHx2/Z/8Ow2Pg/wvHrmqif8A4WN4h0b+0ovChWZFLxL0DJkSE+gXkDJr9Tq5j4r/AAV8IfHbw1/Y/jTwxoPirSw4lW11WyjuokcdGUODhvcc1UJWd2KWqPwl/YS/aC8L+Ef+C5/jLx14o+MVt8UPC2j+AdQuNY8Zz2As7S+jjjiMixxjiRFxsUDcXPAzxXgf7OWqeH/g3+3N4T/ag8QfCnUtF/Zd8SePr2Pw/b3DmS30WTd+7uPKGQRESG4BXKOqklAK/o3H7Hvwl+xQ2x+F/wAO2t7e1FlGjeHLNlSANuEQzH9zdzt6Z561uaj8DPBOr+ALfwnd+DvCt14VsyDb6NNpMD6fBgkjbAU8tcEkjC9Sa19siOQ/DP8A4L06/wCIP2Jf+ChGqfEr4cbbrS/2mPADaM8tm+6K6nMcds7oU4bdF9nYY+8XY15P+3r+zf4q/wCCKN74UsvB0VxLa/Hr4TJ4Y17y2Jzq48sXZTHqzQMo6ne4r+iHWvgF4E8SabotnqPgrwlqFn4bCjSILnR7eWPSguAv2dWQiLG0Y2YxgelW/HXwj8J/FH7D/wAJN4X8O+Iv7Ll8+y/tPTYbz7JJx88fmKdjcDlcHiiNa1kPkPwz+MevS/sI/H39mT4Hx6h4F+CLeHfh9BfXXxV1vQhf3S3Fx9qmnjhY4CL5zSqOeHmYkgYrlv8Agl/+zVo//BQj4+/tq+A7rxVN46t/F2hLcad4lubP7I+o3aXhkt74RHGzMmGH+y3vX7y/Fv8AZ28BfHuysbfxt4N8M+LIdMcyWiatpsV2LUnGdm9TtzgZx1wM1yvxc/Z/Twt8I/Gknwf8K+BvDvxF1jRZNN07UPsMVgpcptiE00UZcxpgEDB+6MCl7XSwuU/FD/gjn8JviJ/wUS/bu+Hvh/4oqbnwh+xvp8to8b5kiuL1bqQW0TZ4Ll0DE9ClkB3Ff0F18j/8Eaf+CdOof8E5v2W7jQfE13p2rePvFGqz6z4j1GzdpY7mdzhAHZVZgqgdQOSxxzX1xU1JXehUVZBRRRWZQUUUUAFFFFABRRRQAUUUUAFFFFABRRRQB//Z"/>
```
Comment (bquenin) [inline pkg/ampmail/accountPasswordConfirm.go:8]: As mentioned during the scrum, emails should not be branded

File: pkg/ampmail/accountPasswordConfirm.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
...f4YuNfvL7z9avpWl07RdNjPl6btbh5ZW5AHQICcg9a9I8EePG1i6t7eOJbnSIb4X1u5TcSCCSPl+pP1r4xuf9ZMNvVAR9OK9y/ZYsNY8V6Jqy2t40i6ZcQKltITtw248fTaBjoR+FbV6cbXOzKcZU9r7OOt9bemrPtjwX8T9L8CeIYZPFHheXUIboqI7qGNGEoz8u85BUjgEg88e9fWn7Nfx18OeNr9tD8H+AtR1TUsBy12dsMD9TlnLYA4yRg8ivjPwB4nt/F+nL4f1m5jKWZdIbaSPl4lw27KnKMB04/Gvpz4VfFC28B+BLWz8Jro+k6rNI1rcX88vm3OGibBwM4w4Q5HODivP9g+W3U+uzLinFKlaFRcnoub7zp/2mZEj+NVlraRs+s6DYzadNLbsVVriRVYBVwDt4YA9lHHAr5e/4KU/BvxB4h/ZI8FfEBZoW8M6Vqs9lJbo+97gSKFS7DjggSxMvOQRJnivoOfwxqnx/+OljHGuoxaG1qjXarL57ytJ8yrBJnKMFLAsMhfMPPTGz/wAF4Ndsvgp/wTm0jwzbxWtrNqFxb6bb2sa7ViQZzgew5/Cs51vY1adKk7yur/qj42nTeIjUxFbZJtPv5n4xzmQYkt1+2QyZUqDzg8HI+lcxretx6Fol3pMf2pbhpFNuxk4htznMTAjP90girei6hILGcbh8pLsoOOtR32rR3li0c0cc/mZYLIu7Z9D2/CvoKlNTVmeDTxM4ydzi0jj53bWNFdVceB7K+2yWt08KsozGwDbT9ciil7Nj9rHqRz2q3chU8bevvUOoWCxxOy8Egk+9FFakKTMe5h/0lV/56Rda+gP+Ceuoeb8Vr7QWXcNd09pEY/djkiG/JHoV3D8aKK48RFOlK57mQyax9Jruvx0Z+i/wI/4J66D+1T4Nvr4Xl1oepQ25X7XazGKV9pY7WIBDDk8kZxj0xWHcfseaL8CpNHk/tbxBrGoRXyu32q9/ccH0VV3fRgQaKK+ZqYmrFtKTPts0y/DOs5ciP0a/Z1+HGj/DnwLY6rHaxtOsAcbVB2gDgAmvx8/4L6ftRal8b/2kNH8JkS2ul+H4zMI2xiSSQlA3XsA3/fQ9KKKjK/fxkebXc+YzSTjQaj6fI+CZLn7PY3kqjl38sD+6BxVSz3eag3H5hlvfiiivs472PlI6p3NW0nyp7dO1FFFUSf/Z"/>
```
Comment (bquenin) [inline pkg/ampmail/accountPasswordConfirm.go:9]: Even though @subfuzion is handsome, I'm not sure we want his face on every single email we send on behalf of amp :)

**Review by bquenin — COMMENTED** [review]
Looks good to me overall!

File: pkg/ampmail/accountPasswordConfirm.go (hunk 2 above)
Comment (freignat91) [reply pkg/ampmail/accountPasswordConfirm.go:9]: see "email design has to be validated, especially the pictures." in the pr body

File: pkg/ampmail/accountPasswordConfirm.go (hunk 1 above)
Comment (freignat91) [reply pkg/ampmail/accountPasswordConfirm.go:8]: see "email design has to be validated, especially the pictures." in the pr body

Comment (bquenin) [reply pkg/ampmail/accountPasswordConfirm.go:8]: Sure, just giving my opinion to open the debate

File: pkg/ampmail/accountPasswordConfirm.go (hunk 2 above)
Comment (bquenin) [reply pkg/ampmail/accountPasswordConfirm.go:9]: Sure, just giving my opinion to open the debate

**Comment by freignat91** [comment]
Close, ampmail is going to be merged on account-users

**Pull request closed without merging**